### PR TITLE
feat: Replace `grpc.io` with PBJ Helidon Plugin

### LINKED ISSUE #180: Use custom gRPC implementation with Helidon
As a Block Node developer
I want to replace Helidon's OOTB gRPC layer with our custom PBJ implementation
So that, I have greater control of the gRPC layer
```[tasklist]
### Tasks
- [x] Replace `server` subproject `grpc.io` gradle dependencies with PBJ's Helidon plugin
- [x] Remove protoc/PBJ Translator code
- [x] Re-implement `grpc.io` integration to align with the Helidon plugin types, etc
- [x] Test our 3 gRPC method types: Unary, Server Streaming, Bidirectional
- [x] Repair Block Node unit tests
- [x] Fix simulator to work with protoc types and remove PBJ types
```

### LINKED ISSUE #311: Flaky unit test
Currently we have a flaky tests, which fails occasionally in CI

```
? testMediatorExceptionHandlingWhenPersistenceFailure()

      org.mockito.exceptions.verification.TooManyActualInvocations: 
      publishStreamResponseObserver2.onNext(
          status {
      }
      
      );
      Wanted 1 time:
      -> at com.hedera.block.server.grpc.BlockStreamServiceIntegrationTest.testMediatorExceptionHandlingWhenPersistenceFailure(BlockStreamServiceIntegrationTest.java:610)
      But was 2 times:
      -> at com.hedera.block.server.producer.ProducerBlockItemObserver.onNext(ProducerBlockItemObserver.java:203)
      -> at com.hedera.block.server.producer.ProducerBlockItemObserver.onEvent(ProducerBlockItemObserver.java:220)
          at app//com.hedera.block.server.grpc.BlockStreamServiceIntegrationTest.testMediatorExceptionHandlingWhenPersistenceFailure(BlockStreamServiceIntegrationTest.java:610)
```

### PR DESCRIPTION
**Pull request by mattp-swirldslabs**
PR Changes:

* Changed from `grpc.io` to `PBJ Helidon Plugin` configuration
* Removed Translator code in the server and simulator
* The server now uses PBJ types exclusively
* The simulator uses protoc types exclusively
* Added NoOp config-based implementations to assist with troubleshooting
* Added new ring buffer metrics and dashboard
* Reduced the default ring buffer sizes based on testing observations
* Added helidon configurability via `helidon.properties`
* Fixed `producer.sh` and `consumer.sh` to work with `repeated` blockitems
* Augmented simulator BlockAsFileLargeDataSets with range config
* Fixes #311
* Fixed tests
* Fixes #180

### CODE REVIEW

**Review by jeromy-cannon — APPROVED** [review]

File: server/src/main/java/com/hedera/block/server/pbj/PbjBlockAccessServiceProxy.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        this.serviceStatus = serviceStatus;
+        this.blockReader = blockReader;
+        this.metricsService = blockNodeContext.metricsService();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @NonNull
```
Comment (AlfredoG87) [inline server/src/main/java/com/hedera/block/server/pbj/PbjBlockAccessServiceProxy.java:77]: are we still using @NonNull annotation on this cases? I thought we are only using it on parameters, constructors and such.

File: server/src/main/java/com/hedera/block/server/pbj/PbjBlockAccessServiceProxy.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        } else {
+            LOGGER.log(ERROR, "Unary singleBlock gRPC method is not currently running");
+
+            return SingleBlockResponse.newBuilder()
+                    .status(SingleBlockResponseCode.READ_BLOCK_NOT_AVAILABLE)
+                    .build();
+        }
+    }
+
+    @NonNull
```
Comment (AlfredoG87) [inline server/src/main/java/com/hedera/block/server/pbj/PbjBlockAccessServiceProxy.java:145]: are we still using @NonNull annotation on this cases? I thought we are only using it on parameters, constructors and such.

File: server/src/main/java/com/hedera/block/server/pbj/PbjBlockAccessServiceProxy.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    private SingleBlockRequest parseSingleBlockRequest(
+            @NonNull final Bytes message, @NonNull final RequestOptions options) throws ParseException {
+        // Copying bytes to avoid using references passed from Helidon
+        return SingleBlockRequest.PROTOBUF.parse(Bytes.wrap(message.toByteArray()));
+    }
+
+    @NonNull
```
Comment (AlfredoG87) [inline server/src/main/java/com/hedera/block/server/pbj/PbjBlockAccessServiceProxy.java:152]: are we still using @NonNull annotation on this cases? I thought we are only using it on parameters, constructors and such.

File: server/src/main/java/com/hedera/block/server/pbj/PbjBlockStreamService.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+         */
+        subscribeBlockStream
+    }
+
+    /**
+     * Streams the block item.
+     *
+     * @return the block item
+     */
+    @NonNull
```
Comment (AlfredoG87) [inline server/src/main/java/com/hedera/block/server/pbj/PbjBlockStreamService.java:55]: are we still using @NonNull annotation on this cases? I thought we are only using it on parameters, constructors and such.

File: server/src/main/java/com/hedera/block/server/pbj/PbjBlockStreamServiceProxy.java [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+    @NonNull
+    private Bytes createSubscribeStreamResponse(
+            @NonNull final SubscribeStreamResponse subscribeStreamResponse, @NonNull final RequestOptions options) {
+        return SubscribeStreamResponse.PROTOBUF.toBytes(subscribeStreamResponse);
+    }
+
+    @NonNull
+    private PublishStreamRequest parsePublishStreamRequest(
+            @NonNull final Bytes message, @NonNull final RequestOptions options) throws ParseException {
+        // Copying bytes to avoid using references from Helidon
```
Comment (AlfredoG87) [inline server/src/main/java/com/hedera/block/server/pbj/PbjBlockStreamServiceProxy.java:194]: could we add here a TODO, to verify if is really necessary and if not to remove the copy bytes in a future PR?

**Review by AlfredoG87 — APPROVED** [review]
Looks really good, this is great job!! 💯 

just leaved a few nits.

File: server/src/main/java/com/hedera/block/server/pbj/PbjBlockAccessService.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    /**
+     * Provides the full name of the BlockStreamService.
+     *
+     * @return the full name of the BlockStreamService.
+     */
+    @NonNull
+    default String fullName() {
+        return "com.hedera.hapi.block." + SERVICE_NAME_BLOCK_ACCESS;
```
[the remark this answers is not in the document]
Comment (mattp-swirldslabs) [reply server/src/main/java/com/hedera/block/server/pbj/PbjBlockAccessService.java:60]: Good catch. I created the constant for that purpose then forgot to fix it. :)

File: server/src/main/java/com/hedera/block/server/pbj/PbjBlockAccessServiceProxy.java (hunk 1 above)
Comment (mattp-swirldslabs) [reply server/src/main/java/com/hedera/block/server/pbj/PbjBlockAccessServiceProxy.java:77]: I thought the convention was to annotate with `@NonNull` if there was a non-primative return type. I see that throughout the codebase.

File: server/src/main/java/com/hedera/block/server/pbj/PbjBlockStreamServiceProxy.java (hunk 5 above)
Comment (mattp-swirldslabs) [reply server/src/main/java/com/hedera/block/server/pbj/PbjBlockStreamServiceProxy.java:194]: Good point. I'll add TODOs for the *Proxy files where applicable

Comment (mattp-swirldslabs) [reply server/src/main/java/com/hedera/block/server/pbj/PbjBlockStreamServiceProxy.java]: Done

**Review by rbarkerSL — APPROVED** [review]
Approval applies to `codecov.yml`

**Review by AlfredoG87 — APPROVED** [review]
LGTM

**Pull request merged**